crates/sui-move/src/build.rs

```diff
@@ -41,6 +41,15 @@ impl Build {
     ) -> anyhow::Result<()> {
         let rerooted_path = base::reroot_path(path.clone())?;
         let build_config = resolve_lock_file_path(build_config, path)?;
+        if let Some(lock_file) = build_config.clone().lock_file {
+            let lock_string = std::fs::read_to_string(lock_file)?;
+            let _compiler_toolchain = move_package::lock_file::schema::CompilerToolchain::read(
+                &mut lock_string.as_bytes(),
+            )?;
+            // TODO: Check for Move.lock compiler toolchain. If it exists and we are not on the current
+            // version, run logic to download and execute the build for a previous version.
+        }
```

Comment:
Intuition: after digging into things it seems reasonable to put the core logic in this function, and update the `Move.lock` at the end of this function body. There may be other suitable places..

Reply:
This may be a little close to the CLI side of the codebase, because the action of publishing or running tests on the package may also trigger a build, and I don't think it goes through this part of the codebase. (The publishing case is probably the most relevant one here).